Day 7. Homework

eclimov opened the pull request:
The following(+ minor improvements) has been implemented:
➢ Paginate jobs on /category/{slug} page
➢ Show 20 jobs per page

eclimov committed:
- Added "knp_paginator" bundle
- Added global variable "jobs_show_limit_category"
- Updated "showJobList" method: the route supports pagination now
- Updated "JobFixtures":
more jobs are being created for debugging purposes;
updated "dateFrom" parameter's value of "dateTimeBetween()" method: there are more chances of getting not-expired job now
- Added "findJobsByCategoryQuery()" method to "JobRepository"
- Added paginator render to "job_list" template

gregurco commented on src/Repository/JobRepository.php:
```diff
@@ -49,4 +51,18 @@ public function findActiveJob(int $id)
+    public function findJobsByCategoryQuery(Category $category): AbstractQuery
+    {
+        return $this->createQueryBuilder('j')
+            ->andWhere('j.category = :category')
```
jobs should be only active

gregurco commented on src/Controller/CategoryController.php:
```diff
@@ -3,28 +3,40 @@
 {
     /**
-     * @Route("/category/{slug}", name="job_list")
+     * @Route("/category/{slug}/{page}", name="job_list", defaults={"page": 1})
```
add requirements for `page`

eclimov committed:
- Added "requirements" attribute to "Route" annotation of "CategoryController"
- Only active jobs are shown on "Category" page now
- Updated "findJobsByCategoryQuery()" method: it's "findActiveJobsByCategoryQuery()" now, - returns query of active jobs only
- Updated "job_list" template

eclimov commented:
Done
